src/parser.js

```diff
@@ -734,20 +734,86 @@ parser.getFileContents = (originalFilename, partialPath) => {
     return { filedata, raw };
 };
 
+const normalizePath = (str) => path.join('', str);
+parser.getSourcesFileContents = (sources, entryFile) => {
+    const included = {};
+    const parentPathOf = (str) => (m = /(.*)\/(?!\/)(.*)\.huff/g.exec(str)) && m[1];
+    const recurse = (filename, parentPath) => {
+        const fileName = normalizePath(filename)
+        const relativePath = path.join(parentPath || '', fileName);
+        let fileString = sources[relativePath];
+        if (!fileString) {
+            if (!fs.existsSync(relativePath)) {
+                throw new Error(`Import not found: ${fileName} ${(parentPath && parentPath !== '.') ? `in ${parentPath}` : ''}`)
+            }
+            fileString = fs.readFileSync(relativePath, 'utf8');
+            console.log(`Warning: File ${relativePath} does not exist in sources, found in filesystem.`)
+        }
+        let formatted = parser.removeComments(fileString);
+        let imported = [];
+        let test = formatted.match(grammar.topLevel.IMPORT);
+        while (test !== null) {
+            const importStatement = formatted.match(grammar.topLevel.IMPORT);
```

Comment:
Your `test` variable will always be up to date, and you don't need to recompute the regex match.